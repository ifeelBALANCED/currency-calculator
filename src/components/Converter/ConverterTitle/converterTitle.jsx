import React from 'react';
import {Heading} from "@chakra-ui/react";

const ConverterTitle = ({title}) => <Heading as="h2" size="xl">{title} </Heading>
export default ConverterTitle;
