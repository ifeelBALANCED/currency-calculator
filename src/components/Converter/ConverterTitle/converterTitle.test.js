import React from "react";
import ConverterTitle from "./converterTitle";

const setUpTitle = (props) => shallow(<ConverterTitle {...props}/>)

describe('should render ConverterTitle component', () => {
    let title
    beforeEach(() => {
        title = setUpTitle()
    })
    // expects that title is wrapped by h2 tag
    it('should contain Heading', () => {
        const wrapper = title.find('Heading')
        expect(wrapper).toHaveLength(1)
    });
    // expects render ConverterTitle component with props
    it('should render ConverterTitle component with props', () => {
        title = shallow(<ConverterTitle title="Some title"/>)
        expect(title).toMatchSnapshot()
    });
    // expects render ConverterTitle component without props
    it('should render ConverterTitle component without props', () => {
        title = shallow(<ConverterTitle/>)
        expect(title).toMatchSnapshot()
    });
})
