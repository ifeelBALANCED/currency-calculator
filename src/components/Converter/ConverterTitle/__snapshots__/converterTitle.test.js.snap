// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConverterTitle component should render ConverterTitle component with props 1`] = `
<Heading
  as="h2"
  size="xl"
>
  Some title
   
</Heading>
`;

exports[`should render ConverterTitle component should render ConverterTitle component without props 1`] = `
<Heading
  as="h2"
  size="xl"
>
   
</Heading>
`;
